lua/metals.lua

```diff
@@ -63,6 +63,47 @@ M.sources_scan = function()
   })
 end
 
+M["metals/quickPick"] = function(_, _, resp)
+  local ids = {}
+  local labels = {}
+  for i, item in pairs(resp.items) do
+    table.insert(ids, item.id)
+    table.insert(labels, i .. '- ' .. item.label )
+  end
+
+  local choice = util.input_list(labels)
+  if (choice == 0) then
+    print("\nmetals: operation cancelled")
+    return { cancelled = true; }
+  else
+    return { itemId = ids[choice] }
+  end
+end
+
+M['metals/inputBox'] = function(_, _, resp)
+    local name = util.input_box(resp.prompt .. ': ')
+
+    if (name == '') then
+      print("\nmetals: operation cancelled")
+      return { cancelled = true; }
+    else
+      return { value = name; }
+    end
+end
+
+M['metals/executeClientCommand'] = function(_, _, cmd_request)
+  if cmd_request.command == 'metals-goto-location' then
+    lsp.util.jump_to_location(cmd_request.arguments[1])
+  end
+end
+
+M.new_scala_file = function(directory_opt)
```

Comment:
In what scenario is the `directory_opt` used here? Is this mainly for someone to use if they want to pass in a parameter?

Reply:
I've added the name_opt as well, so if any of those are passed it won't be requested in the UI. Passing the name is useful for the new symbol quickfix when we create the symbol under cursor.
The directory is probably less useful but it's still an option in the spec. In most cases people will just cd, but I can see -myself if I hace time- using a selection for all the existing packages and selecting destination. Maybe even using a full name: org.corp.NewClass that fills both the folder and the name into one call. Anyway, it doesn't add much code to support it. But feel free to remove if you think it's goldplatting.